server/src/test/java/org/spine3/server/entity/EntityShould.java

```diff
@@ -185,11 +185,81 @@ public void return_id_protobuf_type_name() {
         assertEquals(expected, actual);
     }
 
-    private static Project newProject() {
-        final Project.Builder project = Project.newBuilder()
-                                               .setId(Given.AggregateId.newProjectId())
-                                               .setStatus(Project.Status.CREATED);
-        return project.build();
+    @Test
+    public void generate_non_zero_hash_code() {
+        final int hashCode = entityWithState.hashCode();
```

Comment:
What is wrong with the `hashCode()` returning zero? 

Besides, how do you check all the possible source object states to ensure none of them gives zero as the `hashCode()` invocation result?